lib/DB/Schema/ResultSet/User.pm

```diff
@@ -392,10 +392,10 @@ sub addCourseUser ($self, %args) {
 
 	my $course_user = $self->getCourseUser(info => $args{info}, as_result_set => 1, skip_throw => 1);
 	DB::Exception::UserAlreadyInCourse->throw(
-		message => "The user with " . $args{info}->{username} ? ("username: " . $args{info}->{username})
-		: ("user_id: " . $args{info}->{user_id}) . " is already in the course with " . $args{params}->{course_name}
-		? ("course name: " . $args{info}->{course_name})
-		: ("course_id: " . $args{info}->{course_id}))
+		message => "The user with $args{info}->{username}" ? "username: $args{info}->{username}"
+		: "user_id: $args{info}->{user_id} is already in the course with $args{params}->{course_name}"
+		? "course name: $args{info}->{course_name}"
+		: "course_id: $args{info}->{course_id}")
```

Comment:
@pstaabp:  This is pretty messed up, and needs to be fixed.  The message will always be `"username $args{info}->{username}"` regardless of if `$args{info}->{username}` is defined.